SH4RE/SH4RE/Messages/MessagesChat.swift

```diff
@@ -126,6 +154,138 @@ struct MessagesChat: View {
+                    .padding()
+                
+                Button(action: {
+                    showPopUp.toggle()
+                    
+                    guard let uid = self.vm.chatUser?.uid else{
+                        return
+                    }
+                    
+                    guard let fromId = FirebaseManager.shared.auth.currentUser?.uid else { return }
+                    
+                    getUserName(uid: uid, completion: { name in
+                        
+                        Firestore.firestore().collection("Listings").document(listingId).getDocument { (document, error) in
+                            if let document = document, document.exists {
+                                let data = document.data()!
+                                let ownerId = data["UID"] as? String ?? ""
+                                
+                                var reviewToPost:Review
+                                if ownerId == uid{
+                                    reviewToPost = Review(uid: uid, lid: listingId, rating: Float(reviewRating), description: review, name: name)
+                                }else{
+                                    reviewToPost = Review(uid: uid, lid: "renter", rating: Float(reviewRating), description: review, name: name)
+                                }
+                                Firestore.firestore().collection("messages").document(fromId).collection(uid).whereField("isReviewRequest", isEqualTo: true).getDocuments() {(snapshot, err) in
+                                    snapshot?.documents.forEach({ (document) in
+                                        Firestore.firestore().collection("messages").document(fromId).collection(uid).document(document.documentID).delete()
+                                        var idx = 0
+                                        for message in vm.chatMessages{
+                                            if message.isReviewRequest{
+                                                vm.chatMessages.remove(at: idx)
+                                                idx-=1
+                                            }
+                                            idx+=1
+                                        }
+                                    })
+                                }
+                                Firestore.firestore().collection("recent_messages").document(fromId).collection("messages").document(uid).updateData(["text": "Thanks for leaving a review"])
+                                postReview(review: reviewToPost)
+                            }
+                        }
+                    })
+                })
+                {
+                    Text("Send")
+                }
+                .buttonStyle(primaryButtonStyle())
+                .disabled(review == "" || reviewRating == 0.0)
+                .background(review == "" || reviewRating == 0.0 ? Color.grey : Color.primaryDark)
```

Comment:
this should be placed before button style i think. What I am seeing is:

- button is always primary dark 
- When disabled it is rounded 
- when enabled, the corners are no longer rounded (probably cause setting background to primary dark fills the entire box, even rounded off corners)